Shared/Samples/Realistic lighting and shadows/RealisticLightingAndShadowsView.swift

```diff
@@ -0,0 +1,141 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+import ArcGIS
+
+struct RealisticLightingAndShadowsView: View {
+    /// The view model for this sample.
+    @StateObject private var model = Model()
+    
+    /// A Boolean value indicating whether the settings view should be presented.
+    @State var isShowingSettings = false
+    
+    /// The date second value controlled by the slider.
+    @State private var dateSecond: Float = Model.dateSecondsAfternoon
+    
+    @State private var dateTimeText: String = DateFormatter.localizedString(
+        from: Calendar.current.startOfDay(for: Date()),
+        dateStyle: .medium,
+        timeStyle: .short
+    )
+    
+    /// The atmosphere effect of the scene view.
+    @State private var atmosphereEffect: SceneView.AtmosphereEffect = .realistic
+    
+    /// The sun lighting mode of the scene view.
+    @State private var lightingMode: SceneView.SunLighting = .lightAndShadows
+    
+    /// The sun date that gets passed into the scene view.
+    @State private var sunDate: Date = Calendar.current.startOfDay(for: Date()).advanced(by: TimeInterval(Model.dateSecondsAfternoon))
+    
+    var body: some View {
+        VStack {
+            SceneView(scene: model.scene)
+                .atmosphereEffect(atmosphereEffect)
+                .sunLighting(lightingMode)
+                .sunDate(sunDate)
+            
+            VStack {
+                Slider(value: $dateSecond, in: Model.dateSecondValueRange) {
+                } minimumValueLabel: {
+                    Text("AM")
+                } maximumValueLabel: {
+                    Text("PM")
+                }
+                .frame(maxWidth: 540)
+                .onChange(of: dateSecond, perform: sliderValueChanged(toValue:))
+            }
+            .padding()
+        }
+        .toolbar {
+            ToolbarItemGroup(placement: .bottomBar) {
+                Text(dateTimeText)
+                Button("Mode") {
+                    isShowingSettings = true
+                }
+                .alert("Choose a lighting mode for the scene view.", isPresented: $isShowingSettings) {
+                    ForEach(SceneView.SunLighting?.allCases, id: \.self) { mode in
+                        Button(mode.label) {
+                            lightingMode = mode!
+                        }
+                    }
+                }
+            }
+        }
+    }
+    
+    /// Handle slider value changed event and set the scene view's sun date.
+    /// - Parameter value: The slider's value.
+    private func sliderValueChanged(toValue value: Float) {
+        // A DateComponents struct to encapsulate the minute value from the slider.
+        let dateComponents = DateComponents(second: Int(value))
+        let startOfToday = Calendar.current.startOfDay(for: Date())
```

Comment:
```suggestion
        let startOfToday = Calendar.current.startOfDay(for: .now)
```
Purely a styling choice, I feel this read better. (the `.now` API is added in iOS 15 so wasn't used in the old iOS sample) They are functionally equivalent.

I'm seeing `Calendar.current.startOfDay(for: .now)` get used 3 times. Can you create a variable for it? Wherever you feel it makes the most sense, extension, instance property, view property, etc.